clang/lib/Format/UnwrappedLineParser.cpp

```diff
@@ -2315,6 +2315,8 @@ bool UnwrappedLineParser::tryToParseLambdaIntroducer() {
     if (Next->is(tok::greater))
       return false;
   }
+  if (tok::isLiteral(FormatTok->Tok.getKind()))
+    return false;
```

Comment:
```suggestion
```
Instead, I would change line 2311 to:
```
  if (FormatTok->isOneOf(tok::l_square, tok::numeric_constant))
```

Reply:
With this change this code will crash:
```cpp
Foo foo[] = {
    [0]{1,  1},
    ['a'] { 1, 1 },
    ['a']{1, 1},
};
```

Reply:
With this change, a code like this will crash:
```cpp
Foo foo[] = {
    [0] = {1, 1},
    ['a']{1, 1},
    ['a']{1, 1},
};
```

Reply:
You are right! Then how about the following?
```
  if (FormatTok->is(tok::l_square) || tok::isLiteral(FormatTok->Tok.getKind()))
```
The main point is to return as early as possible.